src/Host/Broker/Impl/Sessions/SessionManager.cs

```diff
@@ -60,13 +60,17 @@ public class SessionManager {
                 }
 
                 session = new Session(this, user, id, interpreter, commandLineArguments);
-                userSessions.Add(session);
             }
 
             session.StartHost(
                 password,
                 _loggingOptions.LogHostOutput ? _hostOutputLogger : null,
                 _loggingOptions.LogPackets ? _messageLogger : null);
+
+            lock (_sessions) {
+                userSessions.Add(session);
```

Comment:
`session.StartHost` is a long operation. Is it possible that there will be concurrent access to the `userSessions`?